packages/compass-e2e-tests/helpers/compass.js

```diff
@@ -303,6 +305,37 @@ function addDebugger(app) {
   }
 }
 
+function wrapCommands(app) {
+  const proto = Object.getPrototypeOf(app.client);
+  const commands = Object.keys(proto).filter((key) => {
+    return (typeof proto[key] === 'function' && !key.includes('.'));
+  });
+
+  const wrapped = {};
+  for (const command of commands) {
+    wrapped[command] = async function (...args) {
+      console.log(`${command} ${args.join(', ')}`);
```

Comment:
Might want to put this behind some flag or setting. Or colourise it, or we could print elapsed durations for the wait style calls or..

But something like it is quite useful for seeing what's going on and getting an idea of where things went wrong. Especially on CI, I think. Without first having to add console logs or somehow trying to reproduce with a debugger.

I really never thought I'd miss this kind of output from nightwatch :)

Reply:
I am not sure how this logging is different from what `addDebugger` method in the same file is doing, it also already puts it "behind a flag" by using the `debug` library and passes arguments to `inspect` which gives them a better stringified representation. You can see the output by running tests with DEBUG=webdriver:client* or by checking the logs of the test run in the CI where the logging is already enabled

If you want to add error stack patching in addition to logging, wouldn't it make sense to do it in a method that already patches the prototype instead of adding another one?

Reply:
I wasn't sure what you would prefer - patching the prototype or adding something new leaving existing methods unchanged. I'm happy with either.

Reply:
Aah `DEBUG=*webdriver:client*` because it is `compass-e2e-tests:webdriver:*`

Reply:
As discussed I'll patch the prototype (probably from the existing method) rather than have wrappedClient.

Reply:
Just for context, we discussed 1/1 and my opinion is that it's totally fine to patch prototype directly if we know what we are doing and it gives us a nicer testing experience.